ui/js/pages/emailList/EmailListModerate.js

```diff
@@ -57,6 +65,14 @@ class EmailList extends Component {
     };
   }
 
+  _toggleContent(messageUri) {
+    return () => {
+      const { collapsed }  = this.state;
+      collapsed[messageUri] = !this._isCollapsed(messageUri);
```

Comment:
The `collapsed` variable should be copied first before being assigned. ReactJS recommends not mutating objects in the state.

```
const collapsed = { ...this.state.collapsed };
...
```

Reply:
Good, thanks.